shell/common/vsync_waiter.cc

```diff
@@ -46,48 +46,87 @@ void VsyncWaiter::AsyncWaitForVsync(Callback callback) {
       return;
     }
     callback_ = std::move(callback);
+    if (secondary_callback_) {
+      // Return directly as `AwaitVSync` is already called by
+      // `ScheduleSecondaryCallback`.
+      return;
+    }
+  }
+  AwaitVSync();
+}
+
+void VsyncWaiter::ScheduleSecondaryCallback(std::function<void()> callback) {
+  FML_DCHECK(task_runners_.GetUITaskRunner()->RunsTasksOnCurrentThread());
+
+  if (!callback) {
+    return;
+  }
+
+  TRACE_EVENT0("flutter", "ScheduleSecondaryCallback");
+
+  {
+    std::scoped_lock lock(callback_mutex_);
+    if (secondary_callback_) {
+      // Multiple schedules must result in a single callback per frame interval.
+      return;
```

Comment:
Shouldn't this at least print out a warning that the callback is getting thrown away?

Reply:
Done. Added a trace event similar to MultipleCallsToVsyncInFrameInterval.